<launch>

    <arg name="project" default="lexicographic_planning"/>

    <!-- Parameters -->
    <rosparam file="$(find lexicographic_planning)/config/params.yaml" command="load" />

    <!--- Run Gazebo -->
    <include file="$(find lexicographic_planning)/launch/include/world.launch" />

    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="$(arg project)_rviz" args="-d $(find lexicographic_planning)/launch/include/rviz.rviz" />

    <!--- Package -->
    <include file="$(find terrain_analysis)/launch/terrain_analysis.launch" />
    
    <include file="$(find terrain_analysis_ext)/launch/terrain_analysis_ext.launch" />
    
    <node pkg="$(arg project)" type="$(arg project)_pathPlanning"   name="$(arg project)_pathPlanning" output="screen" respawn="false" launch-prefix="xterm -e gdb -ex run --args" >
        <param name="planner/steering_angle" value="25.0"/>
        <param name="planner/steering_angle_discrete_num" value="3"/>
        <param name="planner/wheel_base" value="1"/>
        <param name="planner/segment_length" value="1.6"/>
        <param name="planner/segment_length_discrete_num" value="8"/>
        <param name="planner/steering_penalty" value="1.5"/>
        <param name="planner/reversing_penalty" value="3.0"/>
        <param name="planner/steering_change_penalty" value="2.0"/>
        <param name="planner/shot_distance" value="5.0"/>
    </node>

    <node pkg="$(arg project)" type="$(arg project)_pathServer"     name="$(arg project)_pathServer"       output="screen" respawn="false" />
    <node pkg="$(arg project)" type="$(arg project)_cloudRegister"  name="$(arg project)_cloudRegister"    output="screen" respawn="false" launch-prefix="xterm -e gdb -ex run --args"/>
    <node pkg="$(arg project)" type="$(arg project)_obstacleServer" name="$(arg project)_obstacleServer"   output="screen" respawn="false" />
    <node pkg="$(arg project)" type="$(arg project)_pathmap" name="$(arg project)_pathmap"   output="screen" respawn="false" launch-prefix="xterm -e gdb -ex run --args" />

    <!--- Run Move Base -->  
    <!-- <include file="$(find lexicographic_planning)/launch/include/move_base.launch" /> -->


</launch>
